include/LeMonADE/feature/FeatureExcludedVolumeSc.h

```diff
@@ -0,0 +1,424 @@
+template < class IngredientsType>
+bool FeatureExcludedVolumeSc< SpecializedClass<ValueType> >::checkMove( const IngredientsType& ingredients, const MoveAddScMonomer& move ) const
+{
+  if (!latticeFilledUp)
+	throw std::runtime_error("*****FeatureExcludedVolume_T::checkMove....lattice is not populated. Run synchronize!\n");
+  
+  //check if the lattice sites are free
+  VectorInt3 pos=move.getPosition();
+  VectorInt3 dx(1,0,0);
+  VectorInt3 dy(0,1,0);
+  VectorInt3 dz(0,0,1);
+
+  if	(
+	 (ingredients.getLatticeEntry(pos))||
+	 (ingredients.getLatticeEntry(pos+dx))||
+	 (ingredients.getLatticeEntry(pos+dy))||
+	 (ingredients.getLatticeEntry(pos+dx+dy))||
+	 (ingredients.getLatticeEntry(pos+dz))||
+	 (ingredients.getLatticeEntry(pos+dz+dx))||
+	 (ingredients.getLatticeEntry(pos+dz+dy))||
+	 (ingredients.getLatticeEntry(pos+dz+dx+dy))
+	 ) return false;
+  else 
+    return true;
+}
+
+
+/******************************************************************************/
+/**
+ * @fn void FeatureExcludedVolumeSc< SpecializedClass<ValueType> >::applyMove( const IngredientsType& ingredients, const MoveAddScMonomer& move )const
+ * @brief apply excluded volume for MoveAddScMonomer.
+ * 
+ * @param [in] ingredients A reference to the IngredientsType - mainly the system.
+ * @param [in] move A reference to MoveAddScMonomer.
+ * */
+/******************************************************************************/
+template<template<typename> class SpecializedClass, typename ValueType>
+template<class IngredientsType> 
+void FeatureExcludedVolumeSc< SpecializedClass<ValueType> >::applyMove(IngredientsType& ing, const MoveAddScMonomer& move)
+{
+  VectorInt3 pos=move.getPosition();
+  VectorInt3 dx(1,0,0);
+  VectorInt3 dy(0,1,0);
+  VectorInt3 dz(0,0,1);
+
+  ing.setLatticeEntry(pos,1);
+  ing.setLatticeEntry(pos+dx,1);
+  ing.setLatticeEntry(pos+dy,1);
+  ing.setLatticeEntry(pos+dx+dy,1);
+  ing.setLatticeEntry(pos+dz,1);
+  ing.setLatticeEntry(pos+dz+dx,1);
+  ing.setLatticeEntry(pos+dz+dy,1);
+  ing.setLatticeEntry(pos+dz+dx+dy,1);
+}
+
+/******************************************************************************/
+/**
+ * @fn void FeatureExcludedVolumeSc< SpecializedClass<ValueType> >::synchronize(IngredientsType& ingredients)
+ * @brief Synchronizes the lattice occupation with the rest of the system 
+ * by calling the private function fillLattice.
+ *
+ * @param ingredients A reference to the IngredientsType - mainly the system.
+ */
+/******************************************************************************/
+template<template<typename> class SpecializedClass, typename ValueType>
+template<class IngredientsType>
+void FeatureExcludedVolumeSc< SpecializedClass<ValueType> >::synchronize(IngredientsType& ingredients)
+{
+	//note: the lattice entries are set to 0 before by the 
+	//synchronize function of FeatureLattice
+	std::cout << "FeatureExcludedVolumeSc_T::synchronizing lattice occupation...\n";
```

Comment:
To me, adding the "forbidden" moves explicitly to the features looks like the most accurate and safe way to keep an empty/unknown move in the features to enable user specific moves, and at the same time, exclude the wrong lattice move from the excluded volume features. 

Other ideas?